examples/calibration/plot_compare_calibration.py

```diff
@@ -150,48 +156,84 @@ def predict_proba(self, X):
 plt.show()
 
 # %%
-# :class:`~sklearn.linear_model.LogisticRegression` returns well calibrated
-# predictions as it directly optimizes log-loss. In contrast, the other methods
-# return biased probabilities, with different biases for each method:
 #
-# * :class:`~sklearn.naive_bayes.GaussianNB` tends to push
-#   probabilities to 0 or 1 (see histogram). This is mainly
-#   because the naive Bayes equation only provides correct estimate of
-#   probabilities when the assumption that features are conditionally
-#   independent holds [2]_. However, features tend to be positively correlated
-#   and is the case with this dataset, which contains 2 features
-#   generated as random linear combinations of the informative features. These
-#   correlated features are effectively being 'counted twice', resulting in
-#   pushing the predicted probabilities towards 0 and 1 [3]_.
+# :class:`~sklearn.linear_model.LogisticRegressionCV` returns reasonably well
+# calibrated predictions despite the small training set size: its reliability
+# curve is the closest to the diagonal. The reason is two-fold: this estimator
+# directly adjusts its trainable parameter to minimize negative loglikelihood
+# (a.k.a. "log-loss") on a given training set. Furthermore we configured it to
+# automatically tune the `C` regularization parameter to minimize the negative
+# loglikelihood via inner cross-validation.
+#
+# The negative loglikelihood is a proper scoring rule for probabilistic
+# classification models. It means, that in the limit of infinite training data,
+# it is minimized by a model that predicts the true conditional probabilities
+# and would therefore be perfectly calibrated. However, with a finite training
+# set, the calibration can still be suboptimal. Even a large training set,
+# logistic regression models can still be poorly calibrated if they are too
+# regularized or if they are mis-specified (e.g. the true decision boundary is
+# a highly non-linear function of the input features). In this example with a
+# very small training set size the calibration curve of the logistic regression
+# model is closest but not perfectly on the diagonal: this model is slightly
+# under-confident.
```

Comment:
From the calibration plot, logistic regression is slightly under confident on the right side and slightly over confident on the left side.

My definition of "under-confident" is "mean predicted proba < fraction of positives". (I do not think we define the notion of "confidence" in the example)

Reply:
This is what I see on https://github.com/scikit-learn/scikit-learn/pull/28231#issuecomment-1906650776 (the top plot, the one for this PR):

E.g. for mean prediction of 0.8, the frequency of the positively labeled observations for the LR model (green line) is above the diagonal (around 0.9).

The LR models predicts 0.8 instead of 0.9 (on average), hence is under-confident.

Reply:
If I just plot the calibration for logistic regression, I see two regions, one that is over-confident and another that is under-confident:

![Screenshot 2024-02-02 at 4 46 15 PM](https://github.com/scikit-learn/scikit-learn/assets/5402633/2990b50a-558f-4880-b9b6-55ee17ce5375)

Am I missing something here?

Reply:
Ah I see. I would have intuitively both sides as under-confident because on the left side, the models predicts the negative class (with the default decision threshold at 0.5), so in that respect it is under-confident everywhere to me (assuming the curve crosses the diagonal at 0.5).

Not sure what to do. Shall I remove the phrasing under-confident / over-confident completely?

Reply:
In 4fd1215, I defined under-confidence as making predictions too close to 0.5 compared to the fraction of the positive class. I think this is both concise, intuitive and accurate enough.

Reply:
Yea, it makes sense as long as we define what under-confidence means. Is there a reference that gives a standard definition for under-confidence in the context of calibration?

Reply:
What about just „over-estimate“ and „under-estimate“?